### ISSUE TITLE
Support testcase -> mutant mapping in json schema

### ISSUE THREAD

**Issue by Mobrockers** [stryker-mutator/mutation-testing-elements#51]
Since we have coverage analysis, we can show the user which testcases cover which mutants. We would like for the json schema to have support for the coverage analysis mapping. It would look something like this in the resulting json:

```json
{
  "coverage-mapping": [
    {"Unit Test 3": [3, 4, 56, 345]},
    {"Unit Test 4": [3, 8, 45, 113]},
    {"Unit Test 5": [15, 68, 99, 1545]}
  ]
}
```

**Comment by nicojs** [comment]
Could you add a little bit more clarification? What does ` [3, 4, 56, 345]` mean for example. Are those the mutant ids? I think we should call it `mutantCoverage` or something (camelCase). Do we also want to add information about which test killed the mutant? If so, we might need to choose something else than just an array of numbers.

How do you want it to be shown on the screen? We could add it in the modal dialog, that would be the easiest way of showing it I think. 

![image](https://user-images.githubusercontent.com/1828233/61182560-70f39580-a635-11e9-8840-3f3a8c3de58f.png)

Do you also want to communicate to the end-user that all tests were run because he didn't configure coverage analysis? Something like

> Run all 234 tests for this mutant

**Comment by Mobrockers** [comment]
> Could you add a little bit more clarification? What does [3, 4, 56, 345] mean for example. Are those the mutant ids? I think we should call it mutantCoverage or something (camelCase). 

The array is an array of mutant id's. I have no opinion on Schema object naming.

> Do we also want to add information about which test killed the mutant? If so, we might need to choose something else than just an array of numbers.

We do not currently have this information, so we would not be able to provide this detail right now. Also I would add that as a property on the Mutant, and not on this `test case -> mutant mapping` as imo they are separate things.

**Comment by nicojs** [comment]
> We do not currently have this information, so we would not be able to provide this detail right now. Also I would add that as a property on the Mutant, and not on this `test case -> mutant mapping` as imo they are separate things.

In that case, each test case would need an ID as well. How about this:

```json
{
  "tests": [ {
      "id": 1,
      "name": "Foo should bar",
      "mutantCoverage": [42, 1337]
     }, { 
      "id": 1,
      "name": "Baz should qux",    
      "mutantCoverage": []
    }]
}
```

Than on on a mutant: 

```json
{
  "id": "3",
  "killedByTestID": 1,
  "location": {...}
  "replacement": "\"\"",
  "mutatorName": "String Literal",
   "status": "Killed"
},
```

**Comment by nicojs** [comment]
Maybe the id's should be strings, to keep it in line with mutant id's

**Comment by Mobrockers** [comment]
Sorry didn't see these replies. Adding id's for each test case makes sense. We already mostly have them. They should be strings though because a testcase id in the case of dotnet is I think in the form of namespace.testclassname.testmethodname.traitname which I think is required to be unique. In our case it would make little sense not to just reuse that instead of creating a new testcase id definition that would be a number.

**Comment by Mobrockers** [comment]
Closed by #805
